dspblocks/src/main/scala/dsptools/DspBlock.scala

```diff
@@ -176,43 +178,126 @@ abstract class DspBlockTester[V <: DspBlock](dut: V, maxWait: Int = 100)(implici
           bigIntBitsToDouble(y)
         }}).flatten.toSeq
       case _ => 
-        throw DspException(s"Error: packInput: Can't pack input type $gen yet...")
+        throw DspException(s"Error: packInput: Can't unpack output type $gen yet...")
     }
   }
 
+  // unpack complex output data
+  def unpackOutputStream[T<:Data](gen: DspComplex[T], lanesOut: Int): Seq[Complex] = {
+    gen.underlyingType() match {
+      case "SInt" =>
+        streamOut.map(x => (0 until lanesOut).map{ idx => {
+          // TODO: doesn't work if width is > 32
+          val imag = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx)) % pow(2, gen.imaginary.getWidth).toInt
+          val real = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx + gen.imaginary.getWidth)) % pow(2, gen.real.getWidth).toInt
+          Complex(real.toDouble, imag.toDouble)
+        }}).flatten.toSeq
+      case "fixed" =>
+        gen.real.asInstanceOf[FixedPoint].binaryPoint match {
+          case KnownBinaryPoint(binaryPoint) =>
+            streamOut.map(x => (0 until lanesOut).map{ idx => {
+              val imag = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx)) % pow(2, gen.imaginary.getWidth).toInt
+              val real = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx + gen.imaginary.getWidth)) % pow(2, gen.real.getWidth).toInt
+              Complex(toDoubleFromUnsigned(real, gen.real.getWidth, binaryPoint), toDoubleFromUnsigned(imag, gen.imaginary.getWidth, binaryPoint))
+            }}).flatten.toSeq
+          case _ =>
+            throw DspException(s"Error: packInput: Can't create FixedPoint from signal template ${gen.getClass.getName}")
+        }
+      case "real" =>
+        streamOut.map(x => (0 until lanesOut).map{ idx => {
+          // [stevo]: comes out as (imaginary, real) because it's alphabetical
+          val imag = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx))
+          val real = (x >> ((gen.real.getWidth + gen.imaginary.getWidth) * idx + gen.imaginary.getWidth))
+          Complex(bigIntBitsToDouble(real), bigIntBitsToDouble(imag))
+        }}).flatten.toSeq
+      case _ => 
+        throw DspException(s"Error: packInput: DspComplex has unknown underlying type ${gen.getClass.getName}")
+    }
+  }
+
+  // compares chisel and reference outputs, errors if they differ by more than epsilon
   def compareOutput(chisel: Seq[Double], ref: Seq[Double]): Unit = {
     chisel.zip(ref).zipWithIndex.foreach { case((c, r), index) =>
       if (c != r) {
         val epsilon = 1e-12
-        val err = abs(c-r)/abs(r+epsilon)
+        val err = abs(c-r)/(abs(r)+epsilon)
         assert(err < epsilon || r < epsilon, s"Error: mismatch on output $index of ${err*100}%\n\tReference: $r\n\tChisel:    $c")
       }
     }
   }
 
+  // compares chisel and reference outputs, errors if they differ by more than epsilon
+  def compareOutputComplex(chisel: Seq[Complex], ref: Seq[Complex]): Unit = {
```

Comment:
Note that Seq[Double] and Seq[Complex] both reduce to Seq, so if we try to overload one method with the other, Scala complains.